File: src/services/auth-service.ts
import { User, UserRole } from '../types';

// Temporary mock authentication service for development
// This will be replaced with real AWS Amplify authentication once backend is fully set up
export class AuthService {
  private static mockUsers: User[] = [
    {
      id: '1',
      email: 'customer@test.com',
      role: 'customer',
      createdAt: new Date().toISOString(),
      updatedAt: new Date().toISOString(),
    },
    {
      id: '2',
      email: 'admin@test.com',
      role: 'store_admin',
      storeId: 'store1',
      createdAt: new Date().toISOString(),
      updatedAt: new Date().toISOString(),
    }
  ];

  private static currentUser: User | null = null;

  static async signUp(email: string, password: string, role: UserRole, storeName?: string): Promise<void> {
    try {
      console.log('Mock AuthService.signUp called with:', { email, role, storeName });
      
      // Simulate API delay
      await new Promise(resolve => setTimeout(resolve, 1000));
      
      // Check if user already exists
      const existingUser = this.mockUsers.find(user => user.email === email);
      if (existingUser) {
        throw new Error('User already exists');
      }

      // Create new user
      const newUser: User = {
        id: (this.mockUsers.length + 1).toString(),
        email,
        role,
        storeId: role === 'store_admin' ? `store_${Date.now()}` : undefined,
        createdAt: new Date().toISOString(),
        updatedAt: new Date().toISOString(),
      };

      this.mockUsers.push(newUser);
      console.log('Mock signUp completed successfully');
    } catch (error) {
      console.error('Error signing up:', error);
      throw error;
    }
  }

  static async signIn(email: string, password: string): Promise<User> {
    try {
      console.log('Mock AuthService.signIn called with email:', email);

      // Simulate API delay
      await new Promise(resolve => setTimeout(resolve, 1000));

      // For demo purposes, accept any email/password combination
      // In real app, this would validate against Cognito
      let user = this.mockUsers.find(u => u.email === email);
      
      if (!user) {
        // Create a new user if they don't exist (for demo purposes)
        user = {
          id: (this.mockUsers.length + 1).toString(),
          email,
          role: 'customer', // Default to customer
          createdAt: new Date().toISOString(),
          updatedAt: new Date().toISOString(),
        };
        this.mockUsers.push(user);
      }

      this.currentUser = user;
      console.log('Mock signIn completed, user:', user);
      
      return user;
    } catch (error) {
      console.error('Error signing in:', error);
      throw error;
    }
  }

  static async signOut(): Promise<void> {
    try {
      console.log('Mock signOut called');
      this.currentUser = null;
    } catch (error) {
      console.error('Error signing out:', error);
      throw error;
    }
  }

  static async getCurrentUser(): Promise<User | null> {
    try {
      console.log('Mock getCurrentUser called, returning:', this.currentUser);
      return this.currentUser;
    } catch (error) {
      console.error('Error getting current user:', error);
      return null;
    }
  }

  static async isAuthenticated(): Promise<boolean> {
    try {
      return this.currentUser !== null;
    } catch {
      return false;
    }
  }
} 

File: lib/services/auth_service.dart
class AuthService {
  // Hardcoded users with roles
  static const Map<String, Map<String, dynamic>> _users = {
    'user@freshsave.com': {
      'password': 'password123',
      'role': 'customer',
    },
    'admin@freshsave.com': {
      'password': 'admin123',
      'role': 'admin',
    },
  };

  static bool _isLoggedIn = false;
  static String? _currentUser;
  static String? _currentUserRole;

  // Login method
  static bool login(String email, String password) {
    final userData = _users[email];
    if (userData != null && userData['password'] == password) {
      _isLoggedIn = true;
      _currentUser = email;
      _currentUserRole = userData['role'];
      return true;
    }
    return false;
  }

  // Logout method
  static void logout() {
    _isLoggedIn = false;
    _currentUser = null;
    _currentUserRole = null;
  }

  // Check if user is logged in
  static bool get isLoggedIn => _isLoggedIn;

  // Get current user
  static String? get currentUser => _currentUser;

  // Get current user role
  static String? get currentUserRole => _currentUserRole;

  // Check if user is admin
  static bool get isAdmin => _currentUserRole == 'admin';

  // Check if user is customer
  static bool get isCustomer => _currentUserRole == 'customer';

  // Get user display name
  static String get userDisplayName {
    if (_currentUser == null) return '';
    return _currentUser!.split('@')[0];
  }
} 